src/pydash/arrays.py

```diff
@@ -1151,11 +1394,19 @@ def push(array, *items):
         Removed alias ``append``.
     """
     for item in items:
-        array.append(item)
-    return array
+        t.cast(t.List[t.Union[T, T2]], array).append(item)
```

Comment:
Is there a way to separate the casting from the variable modifications? Like do the casting and then do the append as separate statements? I don't necessarily mind the casting if that's necessary, but I think mixing it with the actual function logic could be confusing.

Also, what is the issue that requires casting in the first place? Is this more because of the usage of `T` and `T2` in the return? See my above comment about whether we need to actually drop iterable-item type enforcement in order to support mixed type iterables.

Reply:
The issue here is that even though python allows mixing types in a list, the type system rightfully tells you some thing is wrong if you append a different type to a list declared of a certain type.

```python
# type_test.py
import typing as t
import pydash as _


my_list: t.List[int] = [1]

my_list.append("a")
```
```bash
$ mypy type_test.py
type_test.py:7:16: error: Argument 1 to "append" of "list" has incompatible type "str"; expected "int"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

But for this function I was thinking more flexibility would be nice. The way I typed it, you can pass a list of integers and a string element and the result will be a list of `Union[int, str]` without any error like `append` would report. This means we have to cast the array to bypass the error appending another type would report.

Another way to right this function could be:
```python
    new_array = t.cast(t.List[t.Union[T, T2]], array)

    for item in items:
        new_array.append(item)
    return new_array
```
We would need another intermediate list because just assigning the casted list would report a wrong assignment error.

Let me know if you prefer this approach

Reply:
Another solution would be to ignore type checking too:
```python
    for item in items:
        array.append(item)  # type: ignore
    return array  # type: ignore
```

Reply:
I think I'd be in favor of just ignoring the type checking for cases where casting would be required. I'm more concerned that the input/output typing is correct and less concerned about the internals being 100% on typing if doing so means there's more complexity to get it "right".